### ISSUE TITLE
graph_peak_caller for MC graph

### ISSUE THREAD

**Issue by sdws1983** [cgroza/personalized_genomes_gbio#1]
Hi!

I have come across your impressive work in utilizing graph genomes for ChIP-seq data analysis, including the ChIP analysis in the human pangenome works in [Nature](https://www.nature.com/articles/s41586-023-05896-x). I am very curious about how to perform [graph_peak_caller](https://github.com/uio-bmi/graph_peak_caller/wiki/Graph-based-ChIP-seq-tutorial) analysis on a graph constructed under a [minigraph-cactus](https://github.com/ComparativeGenomicsToolkit/cactus/blob/master/doc/pangenome.md) pipeline. I have attempted various approaches, but none have been successful thus far. [The existing pipelines](https://github.com/uio-bmi/graph_peak_caller/wiki/Creating-vg-graphs) are based on the .fasta + .vcf graphs, and seem not work for the minigraph-cactus graph.

I noticed that you have made the code you used in your Nature article available on [Zenodo](https://doi.org/10.5281/zenodo.6564396). However, I couldn't find information on preprocessing the vg files from minigraph-cactus. This particular step has been a major roadblock for me. If it is convenient for you, could you please provide some guidance or insights?

Thank you very much for your time.

Yumin

**Comment by cgroza** [comment]
Hi,

Indeed, Graph Peak Caller works only with acyclic graphs, so I had to rebuild the minigraph-cactus graph.
I downloaded the VCF file of minigraph-cactus graph, which is publicly available, and filtered for 1-st level snarls with the following script:
https://github.com/glennhickey/pg-stuff/blob/main/strip-nested.py

Then I used vg construct to remake the graph, indexed with vg map and remapped the ChIP-seq data to that.
Then called peaks with Graph Peak Caller.

Hope this helps!
